assignment+4/InvertedIndex.java

```diff
@@ -66,100 +55,97 @@ private int findEntry(String term) {
+    for (int i = 0; i < term.length(); i++) {
+      char c = string.charAt(i);
+      if (c <= '\u007F') {
+        output[counter++] = c;
+      }
+    }
+    String flattenedTerm = (new String(output)).toLowerCase();
+    // step 2 - split the new flattenedTerm, where delimiter = " ", and find the
+    // difference between this new list and the stopWords[] list. The difference
+    // will be our term without stopWords
+    StringBuilder sb = new StringBuilder();
+    for (String word : flattenedTerm.split(" ")) {
+      boolean isWordStopWord = false;
+      for (String stopWord : InvertedIndex.STOP_WORDS) {
+        if (stopWord.equals(word)) {
+          isWordStopWord = true;
           break;
         }
       }
-      if (clean == true) {
-        cleanedTerm = cleanedTerm + w + " ";
+      if(!isWordStopWord){
+        sb.append(word).append(" ");
       }
     }
-    return cleanedTerm.trim();
+    return sb.toString().trim();
   }
 
   // enters a term into the inverted index
-  public void put(String term, Record document) {
+  public void put(String term, T document) {
     String cleanTerm = termCleaner(term);
     int i = findEntry(cleanTerm);
     // if the term doesn't already exist, create a new entry
     if (i == -1) {
-      Record[] documents = new Record[1];
-      documents[0] = document;
-      Entry e = new Entry(cleanTerm, documents);
-      addEntry(e);
+      addEntry(new Entry(cleanTerm, document));
       return;
     }
     /* if term aready exist, add the document to its
     list of documents by creating a new array */
-    Entry e = entries[i];
-    Record[] newDocuments = new Record[e.documents.length + 1];
-    for(int j = 0; j < e.documents.length; j++) {
-      newDocuments[j] = e.documents[j];
-    }
-    newDocuments[newDocuments.length - 1] = document;
-    e.documents = newDocuments;
-    return;
+    this.entries.get(i).documents.add(document);
   }
 
   // gets the documents for a given term
-  public String get(String term) {
+  public ArrayList<T> get(String term) throws NoSuchElementException {
     String cleanTerm = termCleaner(term);
     int i = findEntry(cleanTerm);
     if (i == -1) {
-      String o = "Term was not found";
-      return o;
+      throw new NoSuchElementException();
     }
-    Entry e = entries[i];
-    Record[] records = e.documents;
-    String output = "";
-    for (Record r : records) {
-      output = output + r.getRecordLog() + "\n";
-    }
-    return output;
+    return this.entries.get(i).documents;
   }
 
-  public void printInvertedIndex() {
-    for (Entry e: entries) {
-      for (Record r : e.documents) {
-        System.out.printf("%s: %s\n", e.term, r.cgndbId);
-      }
+  @Override
+  public String toString(){
+    StringBuilder sb = new StringBuilder();
+    for (Entry e : entries){
+      sb.append(e.term + " : " + e.documents.toString())
+        .append("\n");
     }
+    return sb.toString().trim();
   }
 
   public static void main(String[] args) {
```

Comment:
Do we need to comment out all the mains in all the objects?

Reply:
teeeeechnically, there should only ever be one public static void main(String[] args) in our entire program hence why i commented out the other main methods not in Main.java